src/Sarif/Autogenerated/AnnotatedCodeLocationEqualityComparer.cs

```diff
@@ -46,20 +47,20 @@ public bool Equals(AnnotatedCodeLocation left, AnnotatedCodeLocation right)
 
                 foreach (var value_0 in left.Properties)
                 {
-                    string value_1;
+                    SerializedPropertyInfo value_1;
                     if (!right.Properties.TryGetValue(value_0.Key, out value_1))
                     {
                         return false;
                     }
 
-                    if (value_0.Value != value_1)
+                    if (!object.Equals(value_0.Value, value_1))
```

Comment:
Note how the generator knows to call `object.Equals` to compare two `SerializedPropertyInfo` objects, as opposed to the `!=` that it emitted when the dictionary has `string` values. The newly enhanced version of the `DictionaryHint` in the code generator knows how to do that (see the `comparisonType` argument to the `DictionaryHint`s in the code gen hints file later on).